docs/src/user_docs/assembly/io_operations.md

```diff
@@ -37,7 +37,7 @@ In both case the values must still encode valid field elements.
 | --------------- | ----------- | ------------ | ------------------------------------------ |
 | adv_push.*n* <br> - *(n cycles)*   | [ ... ]         | [a, ... ]    | $a \leftarrow stack.pop()$ <br> Pops $n$ values from the advice stack and pushes them onto the operand stack. Valid for $n \in \{1, ..., 16\}$. <br> Fails if the advice stack has fewer than $n$ values. |
 | adv_loadw <br> - *(1 cycle)*     | [0, 0, 0, 0, ... ] | [A, ... ] | $A \leftarrow stack.pop(4)$ <br> Pop the next word (4 elements) from the advice stack and overwrites the first word of the operand stack (4 elements) with them. <br> Fails if the advice stack has fewer than $4$ values. |
-| adv_pipe <br> - *(2 cycles)*     | [S2, S1, S0, a, ... ] | [T2, T1, T0, b, ... ] | $[T_0, T_1, T_2] \leftarrow permute(S_0, stack.pop(4), stack.pop(4))$ <br> $b \leftarrow a + 2$ <br> Pops the next two words (8 elements) from the advice stack, inserts them into memory at address $a$ sequentially, overwrites these top 8 elements onto the operand stack, and performs a Rescue Prime Optimized permutation to the top 12 elements of the operand stack. At the end of the operation, the address is incremented by $2$. <br> Fails if the advice stack has fewer than $8$ values. |
+| adv_pipe <br> - *(2 cycles)*     | [C, B, A, a, ... ] | [F, E, D, a', ... ] | $[D, E, F] \leftarrow permute(A, adv_stack.pop(4), adv_stack.pop(4))$ <br> $a' \leftarrow a + 2$ <br> Pops the next two words from the advice stack, inserts them into memory at address $a$ sequentially. Subsequently, performs a Rescue Prime Optimized permutation using the 2 words of the advice stack as the rate portion of the hasher state, and the deepest word of the operand stack as the capacity (word A), the digest output will be in the deepest word of the operand stack (word E). <br> Fails if the advice stack has fewer than $8$ values. |
```

Comment:
"digest output will be in the deepest word of the operand stack (word E)" should be changed to "digest output will be the second word on the operand stack (word E)."